# Project 27: An interface between Galaxy and disease maps

## Abstract

Analysis of ‘omics’ datasets in biomedicine is challenging due to their high dimensionality and a multitude of necessary processing steps. An important part of the analytical workflow is the interpretation of the results, allowing to select key pathologic agents, enzymes or drug targets. This interpretation can be greatly supported by visualization of the molecules indicated by the analysis in the context of relevant molecular pathways. On the other hand, visualization-based tools require a reproducibility component, allowing to recreate the process of discovery.

Galaxy ([usegalaxy.eu](https://usegalaxy.eu)) is a platform supporting intuitive building of  analytical workflows, and visualizing their logic as flow diagrams. Moreover, it offers reproducible and shareable analytics, as developed pipelines can be published, exchanged and re-run.  Galaxy offers visualization capabilities ([doi:10.1101/842856v3](https://www.biorxiv.org/content/10.1101/842856v3), [galaxyproject.org/learn/visualization/](https://galaxyproject.org/learn/visualization/)), however, these are focused on a level of a single molecule, or genomic landscape. We believe Galaxy will benefit from an interface to a platform capable of visualization of molecular mechanisms, and relationships between molecules. This is particularly interesting because of Galaxy’s capability to handle interactive tools (e.g. [https://live.usegalaxy.eu/](https://live.usegalaxy.eu/)).

Two frameworks for hosting and visualization of molecular interaction diagrams are the MINERVA Platform ([minerva-web.lcsb.uni.lu](https://minerva-web.lcsb.uni.lu)) and WikiPathways ([www.wikipathways.org](www.wikipathways.org)). MINERVA offers large diagram exploration, hierarchical organization of content, data overlay and protein structure viewer. WikiPathways supports large scale crowd-building of pathway diagrams and integration with a range of RDF resources. This way, both solutions complement each other and will benefit from systematic data exchange. In fact, MINERVA and WikiPathways develop interoperability interfaces and work on format exchange in a recently started, joint project on the COVID-19 Disease Map ([https://fairdomhub.org/projects/190](https://fairdomhub.org/projects/190)). Importantly, even though both MINERVA and the WikiPathways-related diagram editor, PathVisio ([https://pathvisio.github.io](https://pathvisio.github.io)), offer client-side plugin architectures, their analytical capabilities cannot match Galaxy workflows, especially considering that omics data require substantial computational power, and frequently need to be run on server side. Introducing the analytical capabilities of the Galaxy platform would greatly increase the value of the diagrams exchanged between the two platforms.

During the Biohackathon, we want to develop an interface between Galaxy, MINERVA and WikiPathways, which will i) bring advanced analytical pipelines to disease and pathway maps, and ii) provide improved interpretation capabilities and an interactive exploration feature to Galaxy workflows. We will build on our existing collaborations, which gives us a significant head start and will allow us to reach a well-developed outcome. We will focus on calculating and visualizing i) risk variants and ii) differentially expressed genes using benchmark datasets to test and validate the interface.

The utility of these Galaxy services will be demonstrated using PancCanNet ([https://www.pancan.org](https://www.pancan.org)), a WikiPathways resource for EU pancreatic cancer translational research projects, aiming to deliver a single resource for collecting, structuring and curating molecular pathways relevant for pancreatic cancer. The outcomes will facilitate and speed up the functional analysis of multi-omics experimental results generated by medical researchers.
 

## Research Area Alignment
This proposal aligns with the “Workflows and workbenches”, “Tools interoperability” and “Galaxy Working Group” tasks of the Tools Platform and Galaxy Community. Moreover, as [demonstrated during the previous Biohackathon](https://github.com/elixir-europe/BioHackathon-projects-2019/blob/master/projects/5/report/report.md) this work aligns with the Rare Diseases Community.

## Topics
- Cancer
- Covid-19
- Galaxy
- Interoperability Platform
- Rare Disease
- Tools Platform

**Project Number:** 27

**EasyChair Number:** 43

## Team

### Lead(s)

1) Marek Ostaszewski, marek.ostaszewski@uni.lu, corresponding author
2) Andrew P. Stubbs, a.stubbs@erasmusmc.nl
3) Chris T. Evelo chris.evelo@maastrichtuniversity.nl

### Nominated participant(s)

- Willem de Koning, Erasmus MC, w.dekoning@erasmusmc.nl  
- Helena Rasche, Erasmus MC, helena.rasche@gmail.com

## Expected outcomes

- A working interface between Galaxy, the MINERVA Platform and WikiPathways
- Two new Galaxy workflows for visualization of analysis results on the diagrams of choice
- New capabilities for two user groups: Galaxy, MINERVA and WikiPathways, for better analytics and visualization of biomedical data.
- An application of the new capabilities using PanCanNet (https://www.pancan.org) to facilitate and speed up the functional analysis of multi-omics experimental results generated by medical researchers.
- A peer-reviewed publication based on the outcomes of the workshop, and finalized afterwards.

## Expected audience

 - Bioinformaticians familiar with Galaxy workflow construction and development of connectors to external APIs
 - Bioinformaticians using Galaxy for genetic variant or transcriptome analysis
 - Bioinformaticians and computational biologists working on pathway and disease maps analysis

**Number of expected hacking days**: 4

