# Project 18: Improve the support of Common Workflow Language in Galaxy (26)

## Abstract

The field of bioinformatics has seen increasing use of the Common Workflow Language (CWL) standards to systematically describe computational pipelines. The 2018 ELIXIR Implementation Study “Enabling the reuse, extension, scaling, and reproducibility of scientific workflows” demonstrated how CWL tool and workflow descriptions can be shared and reused across different workflow executors and workflow management systems (e.g. Toil, Galaxy, CWLEXEC, Cromwell, etc.), addressing the assembly and functional annotation of transcriptomes from isolated marine eukaryotes. One of the main tasks of ...

## Topics

Compute Platfrom
 EOSC-life
 Galaxy
 Interoperability Platform
 Marine Metagenomics
 Tools Platform

**Project Number:** 18



**EasyChair Number:** 26

## Team

### Lead(s)

Nicola Soranzo <nicola.soranzo@earlham.ac.uk>
 Hervé Ménager <herve.menager@pasteur.fr>
 John Chilton <john.chilton@gmail.com>
 Michael Crusoe <mrc@commonwl.org>

### Nominated participant(s)

Marius van den Beek <m.vandenbeek@gmail.com>
 Nicola Soranzo <nicola.soranzo@earlham.ac.uk>

## Expected outcomes

- Implement CWL 1.2 conditionals in Galaxy
 - Advance the merge of the separate branch into the upstream Galaxy repository to be part of the next release of Galaxy
 - Implement a new "Simplified Workflow execution UI" in Galaxy
 - Export Galaxy workflows to non-abstract CWL

## Expected audience

Software developers with either Python or Web Frontend development skills (especially JavaScript/Vue.js), with or without an initial experience of development in Galaxy and/or CWL.

**Number of expected hacking days**: 4

